rasa/shared/core/training_data/story_reader/story_reader.py

```diff
@@ -113,6 +115,14 @@ def _add_event(self, event_name: Text, parameters: Dict[Text, Any]) -> None:
                 "Unknown event '{}'. It is Neither an event "
                 "nor an action).".format(event_name)
             )
+
+        return parsed_events
+
+    def _add_event(self, event_name: Text, parameters: Dict[Text, Any]) -> None:
+        # add 'name' only if event is not a SlotSet,
+        # because there might be a slot with slot_key='name'
```

Comment:
I think these comments can be removed now that you've extracted the code into `_parse_events()`